File: Lab4Cs/Lab4Cs/Trapezoid.cs
public class Trapezoid : Figure
{
    private double a, b, h;

    public Trapezoid() : base() { }

    public Trapezoid(double x1, double y1, double x2, double y2,
                     double a, double b, double h)
        : base(x1, y1, x2, y2)
    {
        this.a = a;
        this.b = b;
        this.h = h;
    }

    public double Area()
    {
        return (a + b) / 2 * h;
    }

    public double Perimeter()
    {
        return a + b + 2 * SideLength();
    }
}

File: Lab4Cpp/Lab4Cpp/Trapezoid.cpp
#include "Trapezoid.h"

Trapezoid::Trapezoid() : Figure(), a(0), b(0), h(0) {}

Trapezoid::Trapezoid(double x1, double y1, double x2, double y2,
    double base1, double base2, double height)
    : Figure(x1, y1, x2, y2), a(base1), b(base2), h(height) {}

Trapezoid::Trapezoid(const Trapezoid& t)
    : Figure(t), a(t.a), b(t.b), h(t.h) {}

double Trapezoid::area() const {
    return (a + b) / 2 * h;
}

double Trapezoid::perimeter() const {
    return a + b + 2 * sideLength();
}